tracim/tracim/model/data.py

```diff
@@ -435,48 +439,516 @@ def toDict(self):
     revision_type = Column(Unicode(32), unique=False, nullable=False, default='')
 
-    parent = relationship('Content', remote_side=[content_id], backref='children')
+    workspace_id = Column(Integer, ForeignKey('workspaces.workspace_id'), unique=False, nullable=True)
+    workspace = relationship('Workspace', remote_side=[Workspace.workspace_id])
+
+    parent_id = Column(Integer, ForeignKey('content.id'), nullable=True, default=None)
+    parent = relationship("Content", foreign_keys=[parent_id], back_populates="children_revisions")
+
+    node = relationship("Content", foreign_keys=[content_id], back_populates="revisions")
     owner = relationship('User', remote_side=[User.user_id])
 
+    """ List of column copied when make a new revision from another """
+    _cloned_columns = (
+        'content_id', 'owner_id', 'label', 'description', 'file_name', 'file_mimetype', 'properties',
+        'file_content', 'type', 'status', 'created', 'updated', 'is_deleted', 'is_archived',
+        'revision_type', 'workspace_id', 'workspace', 'parent_id', 'parent', 'node', 'owner'
+    )
+
+    @classmethod
+    def new_from(cls, revision):
+        """
+
+        Return new instance of ContentRevisionRO where properties are copied from revision parameter.
+        Look at ContentRevisionRO._cloned_columns to see what columns are copieds.
+
+        :param revision: revision to copy
+        :type revision: ContentRevisionRO
+        :return: new revision from revision parameter
+        :rtype: ContentRevisionRO
+        """
+        new_rev = cls()
+
+        for column_name in cls._cloned_columns:
+            column_value = getattr(revision, column_name)
+            setattr(new_rev, column_name, column_value)
+
+        new_rev.updated = datetime.now()
+
+        return new_rev
+
+    def __setattr__(self, key, value):
+        """
+        ContentRevisionUpdateError is raised if tried to update column and revision own identity
+        :param key: attribute name
+        :param value: attribute value
+        :return:
+        """
+        if key in ('_sa_instance_state', ):  # Prevent infinite loop from SQLAlchemy code and altered set
+            return super().__setattr__(key, value)
+
+        if inspect(self).has_identity \
+                and key in self._cloned_columns \
+                and not RevisionsIntegrity.is_updatable(self):
+                raise ContentRevisionUpdateError(
+                    "Can't modify revision. To work on new revision use tracim.model.new_revision " +
+                    "context manager.")
+
+        super().__setattr__(key, value)
+
+    def get_status(self):
+        return ContentStatus(self.status)
+
+    def get_label(self):
+        return self.label if self.label else self.file_name if self.file_name else ''
+
+    def get_last_action(self) -> ActionDescription:
+        return ActionDescription(self.revision_type)
+
+    # Read by must be used like this:
```

Comment:
could you organize source code like this : 
- first, attributes
- then, methods